src/js/components/Box/index.d.ts

```diff
@@ -190,7 +190,13 @@ export interface BoxProps {
   wrap?: boolean | 'reverse';
 }
 
-declare const Box: React.FC<BoxProps & JSX.IntrinsicElements['div']>;
+export interface BoxExtendedProps
+  extends BoxProps,
+    Omit<JSX.IntrinsicElements['div'], keyof BoxProps> {}
```

Comment:
why is `keyof` needed in this case?

Reply:
I noticed `keyof` was already being used in other components, and it is a really nice solution to overiiding keys defined in `JSX.IntrinsicElements`, as it will mean `BoxExtendedProps` doesn't need to be changed if extra keys are ever added to `BoxProps`. This is because `keyof BoxProps` is essentially `key_0 | key_1 | ... | key_n` where `key_i` is a key in the `BoxProps` interface. Adding a new key to `BoxProps` automatically extends this list of keys to exclude.

Reply:
If you would prefer explicit declarations of which keys are being omitted, let me know and I will change `keyof BoxProps` to the union of all offending keys :)

Reply:
Nice. I think it would make sense to use it for use cases that have an overlap of ~3 props.